library/FrontRow/App/Http.hs

```diff
@@ -0,0 +1,122 @@
+-- | Centralized module for making HTTP from the backend
+--
+-- These functions:
+--
+-- - Do not throw exceptions on non-200
+-- - May throw for other 'HttpException' cases (e.g. 'ConnectionTimeout')
+-- - Handle 429-@Retry-In@ for you
```

Comment:
Do we want retry as the default? This seems like something we may want to avoid when making calls within API handlers, etc.

Reply:
This is a tough trade-off. Every time I tried to make this either opt-in or configurable, it really muddied up the interface, working directly against the primary goal of the module.

I could totally be wrong, but my take is that if we have a Handler that is frequently hitting 429's with a long-enough `Retry-In` that it disrupts our own response-time metrics, would be unlikely to decide that "keep calling that 3rd party synchronously within a request, but just don't retry" is the best answer. I think a simple `forkIO` or moving to a Job are where we'd end up. Therefore, I think this behavior is a feature for that use-case -- it'll make the situation louder.

Reply:
That is a reasonable perspective. Just to summarize for my own benefit:

- If retry is taking forever it will inflate the average time of the route.
- We'll notice the route is taking forever.
- We'll likely move the call out to an async process or follow some other mitigation.

Do we have any observability for individual routes?

Reply:
> Do we have any observability for individual routes?

Yup. Request logging has a `route` field, e.g. `SchoolR`, so we can use CloudWatch Insights to query in complex ways that take that into account ("slowest routes" being on naive example).